devtools/client/webconsole/new-console-output/test/mochitest/browser_webconsole_observer_notifications.js

```
@@ -0,0 +1,67 @@
+/* -*- indent-tabs-mode: nil; js-indent-level: 2 -*- */
+/* vim: set ft=javascript ts=2 et sw=2 tw=80: */
+/* Any copyright is dedicated to the Public Domain.
+ * http://creativecommons.org/publicdomain/zero/1.0/ */
+
+"use strict";
+
+const TEST_URI = "data:text/html;charset=utf-8,<p>Web Console test for " +
+                 "notifications";
+
+add_task(function* () {
+  let gotEvents = waitForEvents();
+  let hud = yield openNewTabAndConsole(TEST_URI);
+  yield gotEvents;
+});
+
+function waitForEvents(onConsoleOpened) {
+  function webConsoleCreated(id) {
+    Services.obs.removeObserver(observer, "web-console-created");
+    ok(HUDService.getHudReferenceById(id), "We have a hud reference");
+    content.wrappedJSObject.console.log("adding a log message");
+  }
+
+  function webConsoleDestroyed(id) {
+    Services.obs.removeObserver(observer, "web-console-destroyed");
+    ok(!HUDService.getHudReferenceById(id), "We do not have a hud reference");
+    executeSoon(deferred.resolve);
```

It turns out that this code is firing, and deferred not being declared isn't making anything fail. But looking closer I realized the `webConsoleMessage` code is not firing.

I think this test is probably going to need some additional rewriting from it's existing structure, since things can silently fail.